### PR TITLE
Add walking surface indicators as a new ERSAD field in the facilities

### LINKED ISSUE #1634: ERSAD: support for "walking surface indicators"
As part of the ERSAD fields additions, enabling user to specify what kind of "walking surface indicators" a platform may have. This is a UI representation of the work done in Tiamat's issue #1606, with a specific focus on tactile-related values of the MobilityFacilityList. The UI solution would provide user a way to specify whether any of these conditions are present as part of a certain quay:
- a quay has tactilePlatformEdges and tactileGuidingStrips
- a quay has tactileGuidingStrips
- a quay has tactileGuidingStrips
- a quay doesn't have any tactile features

The XML code behind this can look like this:
```
<facilities>
            <SiteFacilitySet id="FSR:QuaySiteFacilitySet:2453" version="1">
                <MobilityFacilityList> tactilePlatformEdges tactileGuidingStrips</MobilityFacilityList>
            </SiteFacilitySet>
</facilities>
```
The code behind the "walking surface indicators" is meant to update the tactile related values of the MobilityFacilityList.

This is going to be the first example of supporting "facilities" tag in general. More of that is coming in the next issues.

### PR DESCRIPTION
**Pull request by solita-sabinaf**
### Summary

Add new ERSAD field called "walking surface indicators", appearing in the bottom of the facilities tab of a quay.

Screenshot 2025-12-02 at 18 10 57

Screenshot 2025-12-02 at 18 08 49

### Issue

This issue is described in #1634.

Some notes about the implementation:
- First time adding support for a "facilities" field of a quay. To not mix it too much with a "Facilities" tab, changed some of the tab related props types to specifically mention that those related to a tab rather than the "facility" as if a quay-related chunk of model.
- About "facilities" - this is an array of SiteFacilitySet typed items.
- A SiteFacilitySet can contain more fields than just "mobilityFacilityList". More of those will come in the next PR-s.
- A "mobilityFacilityList" can contain other values than just tactile related. The code aims to respect that and update only the tactile related parts of the mobilityFacilityList array.
- The code introduces a new component called FeaturePopoverMenu that can be used in future for any kind of popup menu based ERSAD fields (aka "features" of a station/platform)

Edge cases:
- It is expected that a backend may return null for the facilities altogether (then when attempting to save such stop facilities filed will be initialised with an empty array)
- Or facilities being an empty array
- Or facilities containing one value only - "unknown". In this case, when user chooses to enable a tactile feature, the "unknown" value gets removed as from that on the state of the mobilityFacilityList is not really unknown anymore - it is known that there is something tactile related in the platform.

Closes #1634

### Unit tests

- Manually tested. The edge cases described above were tried out.
- Some update in the existing unit test

### CODE REVIEW

File: src/models/Facilities.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export enum MobilityFacility {
+  TACTILE_PLATFORM_EDGES = "tactilePlatformEdges",
+  TACTILE_GUIDING_STRIPS = "tactileGuidingStrips",
+  UNKNOWN = "unknown",
+  LOW_FLOOR = "lowFloor",
+  STEP_FREE_ACCESS = "stepFreeAccess",
+  SUITABLE_FOR_PUSHCHAIRS = "suitableForPushchairs",
+  SUITABLE_FOR_WHEELCHAIRS = "suitableForWheelchairs",
+  SUITABLE_FOR_HEAVILIY_DISABLED = "suitableForHeaviliyDisabled",
```
Comment (a-limyr) [inline src/models/Facilities.ts]: SUITABLE_FOR_HEAVILIY_DISABLED has a typo - should be "HEAVILY".

**Review by a-limyr — CHANGES_REQUESTED** [review]
The code demonstrates good separation of concerns by creating reusable helpers and components. The naming convention changes (from generic terms to specific "Tab" suffixes) improve code clarity by distinguishing between data model "facilities" and UI "Facilities tabs". Great job!

I have some minor issues and comments about the code.

Approval is pending the corresponding Tiamat PR so I can do some actual testing.

File: src/modelUtils/facilitiesHelpers.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { defaultSiteFacilitySet } from "../models/Facilities";
+import { getIn } from "../utils";
+
+const FacilitiesHelpers = {};
+
+FacilitiesHelpers.getSiteFacilitySet = (entity) => {
+  const facilities = getIn(entity, ["facilities"], []);
+  return facilities[0];
```
[the remark this answers is not in the document]
Comment (solita-sabinaf) [reply src/modelUtils/facilitiesHelpers.js]: Good point! The [] in the preceding line might be a good enough fallback preventing the error, but good to have a the chaining just in case as well. The assumption about "taking the first instance of something" is currently some kind of an agreement made around the ersad fields, similar stuff happening in e.g. place equipments or local services

**Comment by solita-sabinaf** [comment]
@a-limyr, thank you for the review, great points! I have now made the updates, and also Assad has just merged the respective Tiamat's PR - so all ready for another look 🙏
